src/analysis/calculus/inverse.lean

```diff
@@ -496,3 +496,90 @@ theorem to_local_left_inverse {g : 𝕜 → 𝕜} (hg : ∀ᶠ x in 𝓝 a, g (f
 (hf.has_strict_fderiv_at_equiv hf').to_local_left_inverse hg
 
 end has_strict_deriv_at
+
+/-!
+### Inverse function theorem, smooth case
+
+-/
+
+namespace times_cont_diff_at
+variables {E' : Type*} [normed_group E'] [normed_space ℝ E']
+variables {F' : Type*} [normed_group F'] [normed_space ℝ F']
+variables [complete_space E'] (f : E' → F') {f' : E' ≃L[ℝ] F'} {a : E'}
+
+/-- Given a `times_cont_diff` function over `ℝ` with an invertible derivative at `a`, returns a
+`local_homeomorph` with `to_fun = f` and `a ∈ source`. -/
+def to_local_homeomorph
+  {n : with_top ℕ} (hf : times_cont_diff_at ℝ n f a) (hf' : has_fderiv_at f (f' : E' →L[ℝ] F') a)
+  (hn : 1 ≤ n) :
+  local_homeomorph E' F' :=
```

Comment:
If you prove
```lean
lemma times_cont_diff_at.has_strict_fderiv_at' (hf : times_cont_diff_at ℝ n f a) (hf' : has_fderiv_at f f' a) (hn : 1 ≤ n) :
  has_strict_fderiv_at f f' a :=
by simpa only [hf'.fderiv] using hf.has_strict_fderiv_at
```
then this can becone `(hf.has_strict_fderiv_at' hf' hn).to_local_homeomorph`. This should save you some `rw`s in the proofs. You can also define `times_cont_diff_at.local_inverse` as `(hf.has_strict_fderiv_at' hf' hn).local_inverse`, then lemma `rename_me` can be `times_cont_diff_at.to_local_inverse` mirroring `has_strict_fderiv_at` API.

Reply:
Great, thanks for this!